src/api/client/Assets.ts

```diff
@@ -250,4 +257,77 @@ export class Assets {
       next,
     };
   }
+
+  /**
+   * Retrieve all the Asset Global Metadata on chain.
+   * This includes metadata id, name and optional specs
+   */
+  public async getGlobalMetadata(): Promise<AssetGlobalMetadata[]> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            asset: { assetMetadataGlobalKeyToName, assetMetadataGlobalSpecs },
+          },
+        },
+      },
+    } = this;
+
+    const [keyToNameEntries, specsEntries] = await Promise.all([
+      assetMetadataGlobalKeyToName.entries(),
+      assetMetadataGlobalSpecs.entries(),
+    ]);
+
+    return keyToNameEntries.map(
+      ([
+        {
+          args: [rawId],
+        },
+        rawName,
+      ]) => {
+        const metadata: AssetGlobalMetadata = {
+          id: u64ToBigNumber(rawId),
+          name: bytesToString(rawName.unwrap()),
+        };
+
+        const specEntry = specsEntries.find(
+          ([
+            {
+              args: [rawKeyId],
+            },
+          ]) => rawKeyId.eq(rawId)
+        );
+
+        if (specEntry) {
+          const [, rawSpecs] = specEntry;
+          if (rawSpecs.isSome) {
+            const specs: AssetMetadataSpec = {};
+            const {
+              url: rawUrl,
+              description: rawDescription,
+              typeDef: rawTypeDef,
+            } = rawSpecs.unwrap();
+
+            if (rawUrl.isSome) {
+              specs.url = bytesToString(rawUrl.unwrap());
+            }
+
+            if (rawDescription.isSome) {
+              specs.description = bytesToString(rawDescription.unwrap());
+            }
+
+            if (rawTypeDef.isSome) {
+              specs.typeDef = bytesToString(rawTypeDef.unwrap());
+            }
+
+            if (Object.keys(specs).length > 0) {
```

Comment:
I think its OK to assign an empty object to specs. Let me know if there is something I am not considering.